### ISSUE TITLE
Assistance Required: Configuring Secure Connection Establishment

### ISSUE THREAD

**Issue by crapulent** [Cinegy/Cinegy.SRT#3]
First of all, thank you for providing this fantastic NuGet package!

I have encountered an issue while trying to set up a secure connection establishment using the guide available at https://github.com/Haivision/srt/blob/master/docs/features/access-control.md. In order to achieve this, I need to call the srt.srt_listen_callback method and pass a delegate for handling the connection before it is fully established.

I am not very experienced with SWIG, but from what I understand, the auxiliary delegate types should be defined before package generation, which I cannot do since the source code is not accessible. Could you please provide guidance on the proper course of action in this situation?

**Comment by BlackGad** [comment]
@crapulent Thank you for kind words!

New package version will be released very soon, which will solve the issues you have encountered. The new version will include the necessary auxiliary delegate types and a lot of additional type mappings.

Thank you for bringing this issue to our attention!

Spoiler code from new API related to your issue:
```csharp
using var socket = srt.srt_create_socket().Validate();
srt.srt_setsockflag_bool(socket, SRT_BOOL_SOCKOPT.SRTO_BOOL_RCVSYN, false).Validate("SRT unblock socket");

var bindingEndPoint = new IPEndPoint(IPAddress.Any, options.SrtPort);
srt.srt_bind(socket, bindingEndPoint).Validate("Bind listen socket");

void ListenCallback(IntPtr opaque, int ns, int version, IPEndPoint address, string id)
{
    Console.WriteLine($"CALLBACK: SOCKET for stream {address} with '{id}' was accepted");

    srt.srt_setsockflag_string(ns, SRT_STRING_SOCKOPT.SRTO_STRING_PASSPHRASE, "some password");
}

srt.srt_listen_callback(socket, ListenCallback, IntPtr.Zero).Validate("Setting listen callback");
srt.srt_listen(socket, 5).Validate("Setting listen backlog");

Console.WriteLine($"SRT client connected {bindingEndPoint}");
Console.WriteLine($"Waiting for SRT connection on {options.SrtPort}");
```

**Comment by crapulent** [comment]
Thank you, this looks awesome and incredibly convenient. Special thanks for the ability to pass strings in srt_setsockflag_string. I'm eagerly looking forward to the next release.

**Comment by BlackGad** [comment]
@crapulent Examples updated. This examples based on alpha version of [SrtSharp](https://www.nuget.org/packages/SrtSharp/1.5.1.10038-alpha) package. Have fun :)

**Comment by crapulent** [comment]
sorry for the long delay. I tried the new lib version, and everything works great and is very convenient. It functioned flawlessly on Windows and inside Linux Docker containers. Thanks very much again :). 

However, I encountered an issue when running a Linux amd64 container on a MacBook M1. I understand that this use case might not have been considered and might not be a priority to support, but I wanted to bring it up just in case it indirectly indicates other potential problems that could arise on different platforms.

The issue: The srt_accept call never completes - neither when attempting to connect nor after executing srtSocket.Dispose(); srt.srt_cleanup();. The application version with the previous library version worked correctly under these conditions.

Again, I'm not asking for a fix, as this is likely a rare use case that may not be necessary for production environments. I'm just notifying you in case it is relevant, and if needed, I'm willing to provide an example for reproducing the issue.

P.S. This issue can be closed, as my problem is resolved. Thank you!
